app/src/services/gelato.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  Action as GelatoAction,
+  Condition as GelatoCondition,
+  GelatoProvider,
+  Operation,
+  Task,
+  TaskReceipt,
+} from '@gelatonetwork/core'
+import { Contract, Wallet, constants, ethers, utils } from 'ethers'
+
+import { GELATO_MIN_USD_THRESH } from '../common/constants'
+import { getLogger } from '../util/logger'
+import { getToken } from '../util/networks'
+import { GelatoData } from '../util/types'
+
+const logger = getLogger('Services::GelatoService')
+
+const gelatoCoreAbi = [
+  'function submitTask(tuple(address addr, address module) _provider, tuple(tuple(address inst, bytes data)[] conditions, tuple(address addr, bytes data, uint8 operation, uint8 dataFlow, uint256 value, bool termsOkCheck)[] actions, uint256 selfProviderGasLimit, uint256 selfProviderGasPriceCeil) _task, uint256 _expiryDate)',
+  'function cancelTask(tuple(uint256 id, address userProxy, tuple(address addr, address module) provider, uint256 index, tuple(tuple(address inst, bytes data)[] conditions, tuple(address addr, bytes data, uint8 operation, uint8 dataFlow, uint256 value, bool termsOkCheck)[] actions, uint256 selfProviderGasLimit, uint256 selfProviderGasPriceCeil)[] tasks, uint256 expiryDate, uint256 cycleId, uint256 submissionsLeft) _TR)',
+]
+
+const actionWithdrawLiquidutyAbi = [
```

Comment:
Can you fix the typo please? Should be 'actionWithdrawLiquidityAbi'

Reply:
done